xskillscore/core/probabilistic.py

```diff
@@ -475,6 +478,64 @@ def threshold_brier_score(
         return res.mean(dim, keep_attrs=keep_attrs)
 
 
+def _check_bin_dim(ds):
+    """Assert that bin_dim is in ds. Try to guess and rename edges dimension."""
+    bin_dim = "category_edge"
+    if bin_dim not in ds.dims:
+        raise ValueError(f"require {bin_dim} dimension, found {ds.dims}")
+    return ds
+
+
+def _check_data_within_edges(forecasts, forecasts_edges):
+    """Check that forecasts_edges has category_edge dimension and forecasts_edges span
+    range of forecasts (per variable if xr.Dataset), otherwise ValueError."""
+    if "category_edge" not in forecasts_edges.dims:
+        raise ValueError(
+            "Expect to find category_edge in forecast_edges, found "
+            f"{forecasts_edges.dims}"
+        )
+
+    def _helper_check_data_within_edges(forecasts, forecasts_edges):
+        if forecasts.min() < forecasts_edges.min():
+            raise ValueError(
+                "found forecasts outside forecast_edges, found forecasts_edges.min() = "
+                f"{forecasts_edges.min()} and forecasts.min() = {forecasts.min()}"
+            )
+        if forecasts.max() > forecasts_edges.max():
+            raise ValueError(
+                "found forecasts outside forecast_edges, found forecasts_edges.max() = "
+                f"{forecasts_edges.max()} and forecasts.max() = {forecasts.max()}"
+            )
+
+    if isinstance(forecasts, xr.Dataset):
+        for v in forecasts.data_vars:
+            _helper_check_data_within_edges(forecasts[v], forecasts_edges[v])
+    elif isinstance(forecasts, xr.DataArray):
+        _helper_check_data_within_edges(forecasts, forecasts_edges)
+    else:
+        raise ValueError("only defined for xr.DataArrays and xr.Datasets")
+
+
+def _raise_value_error_cdf(cdf):
+    """Check basic characteristics of a cumulative distribution function."""
+
+    def helper_raise_value_error_cdf(cdf):
+        # CDF <=1
+        if not (cdf <= 1.0).all():
```

Comment:
I think this will trigger dask computation. Perhaps you could only run these if not a dask array?

Reply:
Probably I don’t need to run this. Was a safety net while developing.